src/transformStateWithClones.js

```diff
@@ -6,8 +6,40 @@
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
   // write code here
+  const result = [];
+  let copy = { ...state };
+
+  for (const object of actions) {
+    switch (object.type) {
+      case 'addProperties':
+        copy = Object.assign({}, copy, object.extraData);
+        break;
+
+      case 'removeProperties':
+        copy = removeProperties(copy, object.keysToRemove);
+        break;
+
+      case 'clear':
+        copy = {};
```

Comment:
Assigning `copy` to an empty object is correct for the 'clear' action. This ensures that `copy` is a new object.